### ISSUE TITLE
git.detached with specified rev TypeError: object of type 'NoneType' has no len()

### ISSUE THREAD

**Issue by stamak** [saltstack/salt#42664]
### Description of Issue/Question

git.detached with specified rev TypeError: object of type 'NoneType' has no len()

```
local:
----------
          ID: salt_clone
    Function: git.detached
        Name: https://github.com/saltstack/salt.git
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1837, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1794, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python2.7/dist-packages/salt/states/git.py", line 2216, in detached
                  if len(ref) <= 40 \
              TypeError: object of type 'NoneType' has no len()
     Started: 12:09:13.651527
    Duration: 2.62 ms
     Changes:
```
### Setup

```
# states/git.sls
salt_clone:
  git.detached:
    - name: https://github.com/saltstack/salt.git
    - rev: 4efced05315aac81a12f6020f0cd7bcb01993da1
    - target: /root/salt_git
```

### Steps to Reproduce Issue

```sudo salt-call state.apply git```

### Versions Report
```
salt --versions-report
Salt Version:
           Salt: 2017.7.0

Dependency Versions:
           cffi: 1.10.0
       cherrypy: Not Installed
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: 2.18
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 19 2016, 06:48:10)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4


System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-87-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

**Comment by rickh563** [comment]
ZD-1643

**Comment by rickh563** [comment]
This Issue appears to be a duplicate of https://github.com/saltstack/salt/issues/42381 . 
This needs to be validated.

**Comment by gtmanfred** [comment]
That is the same issue, and is fixed in 2017.7.1
